src/Messages.sol

```diff
@@ -159,8 +159,8 @@ library ConnectorMessages {
      * 47-78: amount (uint256 = 32 bytes)
      * 
      */
-    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint256 amount) internal pure returns (bytes memory) {
-        return abi.encodePacked(uint8(Call.Transfer), poolId, trancheId, user, amount);
+    function formatTransfer(uint32 destinationDomain, uint64 poolId, bytes16 trancheId, address user, uint256 amount) internal pure returns (bytes memory) {
```

Comment:
Let's move this argument to the end so that it matches with the encoding order 

```suggestion
    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint256 amount, uint32 destinationDomain) internal pure returns (bytes memory) {
```